.github/workflows/reusable-release.yml

```diff
@@ -59,6 +61,25 @@ jobs:
           key_fingerprint=$key_fingerprint
           key_email=$key_email
           EndOfFile
+      - name: Check for default branch
+        id: generate-version-suffix
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          default_branch=$(gh repo view --json defaultBranchRef --jq .defaultBranchRef.name ${{ github.repository }})
+   
+          if [[ "$default_branch" = $GITHUB_REF_NAME ]]; then
+            release_type="FULL_MAIN_BRANCH"
+            version_suffix=""
+          else
+            release_type="PREVIEW_FEATURE_BRANCH"
+            version_suffix="-PREVIEW.${GITHUB_REF_NAME//[^[:alnum:-_]]/}.$(date +%Y-%m-%dT%H%M).${GITHUB_SHA:0:8}"
```

Comment:
Lots of different stuff going on in the `version_suffix`! :

* Starting `-` hyphen - https://semver.org/#spec-item-9 says _"9. A pre-release version MAY be denoted by appending a hyphen ..."_
* `PREVIEW` - this is recognised by [IntelliJ](https://github.com/JetBrains/intellij-community/blob/2370fcc6840c5bd39a90ba0c4e71466d8be92ec3/plugins/package-search/src/com/jetbrains/packagesearch/intellij/plugin/ui/toolwindow/models/versions/PackageVersionNormalizer.kt#L45-L72), [Scala Steward](https://github.com/scala-steward-org/scala-steward/blob/0ab0c50e369ab486c4f2402aad35add36ad16fee/modules/core/src/main/scala/org/scalasteward/core/data/Version.scala#L167-L181) to mean _'non-stable-release, not suitable for auto-upgrading to'_, which is what we want 👍 
* `${GITHUB_REF_NAME//[^[:alnum:-_]]/}` - this is the PR branch name, with most non-alphanumeric characters stripped out - in particular, stripping out slash (`/`), which some people use in Git branch names but which we couldn't have as part of the version, because Maven uses the version as part of the artifact path in the url.
* `$(date +%Y-%m-%dT%H%M)` - this is the date, to the minute, helping to differentiate and order mulitple preview releases for one PR.
* `${GITHUB_SHA:0:8}` - this is the first 8 digits of the hash id for the PR commit that the preview release was based on. This helps to differentiate and identify multiple preview release for one PR, but also is recognised by [Scala Steward](https://github.com/scala-steward-org/scala-steward/blob/0ab0c50e369ab486c4f2402aad35add36ad16fee/modules/core/src/main/scala/org/scalasteward/core/data/Version.scala#L96-L114), again meaning _'non-stable-release, not suitable for auto-upgrading to'_, which is what we want 👍